src/main/java/org/janelia/saalfeldlab/n5/hdf5/N5HDF5Reader.java

```diff
@@ -340,25 +340,49 @@ public <T> T getAttribute(String pathName, final String key, final Class<T> claz
 				return (T)reader.int64().getArrayAttr(pathName, key);
 			else
 				return (T)reader.uint64().getArrayAttr(pathName, key);
-		if (type.isAssignableFrom(int[].class))
+		else if (type.isAssignableFrom(long[][].class))
+			if (attributeInfo.isSigned())
+				return (T)reader.int64().getMatrixAttr(pathName, key);
+			else
+				return (T)reader.uint64().getMatrixAttr(pathName, key);
+		else if (type.isAssignableFrom(int[].class))
 			if (attributeInfo.isSigned())
 				return (T)reader.int32().getArrayAttr(pathName, key);
 			else
 				return (T)reader.uint32().getArrayAttr(pathName, key);
-		if (type.isAssignableFrom(short[].class))
+		else if (type.isAssignableFrom(int[][].class))
```

Comment:
Just to be sure, are `int[][]` assignable from `int[]`? If so, the 2D array variants should come before the 1D arrays.

Reply:
:+1: good catch, yes we need to check that.

Reply:
Turns out
```
int[].class.isAssignableFrom( int[][].class ); // returns false
int[][].class.isAssignableFrom( int[].class ); // returns false
```